tests/epyccel/test_epyccel_functions.py

```diff
@@ -12,6 +12,15 @@ def clean_test():
     shutil.rmtree('__pycache__', ignore_errors=True)
     shutil.rmtree('__epyccel__', ignore_errors=True)
 
+
+def test_func_no_args_f1():
+    def f1():
+        from numpy import pi
+        value = (2*pi)**(3/2)
```

Comment:
Do we know why this failed? It seems like the sympy inference mechanism ought to have been sufficient for a constant expression

Reply:
we don't evaluate the constant expression , so sympy treats it as a normal expression

Reply:
Yes I realise that, but what I mean is that sympy ought to have been capable of treating it. There shouldn't be any pyccel objects that behave differently than the equivalent sympy object would have done should there?

The last time we had a problem like this it was due to mishandling of Variables, and sympy being unable to tell if something was different from zero. Here we can't have anything like that as everything is constant

Reply:
I understand that Pow was not able to determine whether its result was `integer`, `real` or `complex`, and that is now fixed. But can anybody explain why we were getting a `RecursionError`?

Reply:
yes I'm not sure what was the problem i just added the types by myself in those methods so that sympy doesn't do it by himself

Reply:
The `RecursionError` is due to sympy bouncing between 2 functions when analysing the type. Sympy's type inference mechanism is much more sophisticated than we need and checks things such as whether the exponent is negative or 0. This kind of information is not provided by pyccel.

Sympy's type inference mechanism seems quite sensitive to any missing information and seems to fall into infinite loops between 2 functions in this case (this is the second time we have seen this). This is why it is important for us to move away from sympy's type inference (see issue #305 )